Makefile

```diff
@@ -40,30 +74,47 @@ guestvm:
 tellus: guestvm
 	cargo build --package test_workloads --bin tellus --release
 
+# Runnable targets:
+#
+#  run_tellus_gdb: Run Tellus as the host VM with GDB debugging enabled.
+#  run_tellus: Run Tellus as the host VM.
+#  run_linux: Run a bare Linux kernel as the host VM.
+#  run_debian: Run a Linux kernel as the host VM with a Debian rootfs.
+
 run_tellus_gdb: tellus_bin salus_debug
-	     ${LOCAL_PATH}qemu-system-riscv64 \
-		     -s -S \
-		     ${MACH_ARGS} -smp ${NCPU} -m ${MEM_SIZE} \
-		     -nographic \
-		     -kernel target/riscv64gc-unknown-none-elf/debug/salus \
-		     -device guest-loader,kernel=tellus_guestvm,addr=0xc0200000
+	$(QEMU_BIN) \
+		-s -S $(MACH_ARGS) \
+		-kernel $(DEBUG_BINS)salus \
+		-device guest-loader,kernel=tellus_guestvm,addr=$(KERNEL_ADDR) \
+		$(EXTRA_QEMU_ARGS)
 
 run_tellus: tellus_bin salus
-	     ${LOCAL_PATH}qemu-system-riscv64 \
-		     ${GDB_ARGS} \
-		     ${MACH_ARGS} -smp ${NCPU} -m ${MEM_SIZE} \
-		     -nographic \
-		     -kernel target/riscv64gc-unknown-none-elf/release/salus \
-		     -device guest-loader,kernel=tellus_guestvm,addr=0xc0200000
+	$(QEMU_BIN) \
+		$(MACH_ARGS) \
+		-kernel $(RELEASE_BINS)salus \
+		-device guest-loader,kernel=tellus_guestvm,addr=$(KERNEL_ADDR) \
+		$(EXTRA_QEMU_ARGS)
 
 run_linux: salus
-	     ${LOCAL_PATH}qemu-system-riscv64 \
-		     ${GDB_ARGS} \
-		     ${MACH_ARGS} -smp ${NCPU} -m ${MEM_SIZE} \
-		     -nographic \
-		     -kernel target/riscv64gc-unknown-none-elf/release/salus \
-		     -device guest-loader,kernel=../linux/arch/riscv/boot/Image,addr=0xc0200000 \
-		     -append "console=ttyS0 earlycon=sbi keep_bootcon bootmem_debug"
+	$(QEMU_BIN) \
+		$(MACH_ARGS) \
+		-kernel $(RELEASE_BINS)salus \
+		-device guest-loader,kernel=$(LINUX_BIN),addr=$(KERNEL_ADDR) \
+		-append "$(BOOTARGS)" \
+		$(EXTRA_QEMU_ARGS)
+
+run_debian: salus
+	$(QEMU_BIN) \
+		$(MACH_ARGS) \
+		-kernel $(RELEASE_BINS)salus \
+		-device guest-loader,kernel=$(LINUX_BIN),addr=$(KERNEL_ADDR) \
+		-append "$(BOOTARGS) root=LABEL=rootfs" \
+		-device guest-loader,initrd=$(INITRD_IMAGE),addr=$(INITRD_ADDR) \
+		-device x-riscv-iommu-pci \
+		-drive file=$(ROOTFS_IMAGE),if=none,id=hd \
+		-device nvme,serial=deadbeef,drive=hd \
+		-netdev user,id=usernet,hostfwd=tcp:127.0.0.1:7722-0.0.0.0:22 \
+		-device e1000e,netdev=usernet
```

Comment:
nit pick: just stick $(EXTRA_QEMU_ARGS) for debian as well ?

Reply:
Ah whoops, intended that to be there to begin with. Added.